src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -269,7 +279,10 @@ struct FindComponentNearestNeighbors
             if (candidate_edge < current_best)
             {
               current_best = candidate_edge;
-              Kokkos::atomic_min(&radius, candidate_dist);
+              if constexpr (SharedRadii == SharedRadiiPolicy::ATOMIC)
+                Kokkos::atomic_min(&radius, candidate_dist);
```

Comment:
How much slower is it with atomics?

Reply:
The cost of serial atomics. I have not explicitly measured it.